### ISSUE TITLE
Focus and Bang are broken for DescribeSpec

### ISSUE THREAD

**Issue by nikitabobko** [kotest/kotest#1047]
# Focus
Focus `Foo` describe closure:
```kotlin
class Foo : DescribeSpec({
  describe("f:Foo") {
    it("foo") {
      assertEquals(1, 2)
    }
  }

  describe("Bar") {
    it("bar") {
      assertEquals(1, 2)
    }
  }

  describe("Baz") {
    it("baz") {
      assertEquals(1, 2)
    }
  }
})
```
Expected: only `describe Foo` and `it foo` executed
Actual: `foo` ignored. `Bar` and `Baz` are executed and didn't fail?!
![image](https://user-images.githubusercontent.com/20517828/67250387-db597480-f473-11e9-97bc-bc4bc1a4f48d.png)

# Bang
```kotlin
class Foo : DescribeSpec({
  describe("!Foo") {
    it("foo") {
      assertEquals(1, 2)
    }
  }

  describe("Bar") {
    it("bar") {
      assertEquals(1, 2)
    }
  }

  describe("Baz") {
    it("baz") {
      assertEquals(1, 2)
    }
  }
})
```
Expected: `Foo` is skipped. `Bar` and `Baz` failed
Actual: `Bar` and `Baz` failed. `Foo` is passed?
![image](https://user-images.githubusercontent.com/20517828/67251199-ea8df180-f476-11e9-8283-2d0a2f9d70e8.png)

**Comment by LeoColman** [comment]
A first analysis tells me that the test is being correctly ignored (if you throw an exception inside it, for example, it won't fail), but it's marked incorrectly in IntelliJ.

This for both scenarios as per some tests I made.

I'm trying to see if this has anything to do with `DescribeSpec` itself or the way we show IntelliJ the results

**Comment by sksamuel** [comment]
Fixed in master for 4.0
